probe-rs/src/core/mod.rs

```diff
@@ -531,6 +531,7 @@ pub enum Architecture {
 pub enum CoreStatus {
     Running,
     Halted(HaltReason),
+    LockedUp,
```

Comment:
Please add a comment here, to indicate that this only happens for Cortex-M processors, not for RISCV.